src/lib/util.ts

```diff
@@ -106,38 +90,105 @@ export const NAME_PROHIBITED_ATTRIBUTES = new Set<string>([
   'aria-roledescription',
 ] satisfies NameProhibitedAttributes[]);
 
+const LANDMARK_ROLES = Object.keys(landmarkRoles) as LandmarkRole[];
+const LANDMARK_ELEMENTS: TagName[] = ['article', 'aside', 'main', 'nav', 'section'];
+const LANDMARK_CSS_SELECTOR = LANDMARK_ROLES.map((role) => `[role=${role}]`)
+  .concat(...LANDMARK_ELEMENTS.map((el) => `${el}:not([role])`))
```

Comment:
Here, we want to check for `main` as a parent, but NOT `main[role="presentation"]`, so we eliminate anything with a `role` set. Fortunately, we are checking for `[role="main"]` from the roles list, too, so it should find all scenarios correctly, in a compact CSS selector.

We only have these mapping functions to make sure in a list of 10+ selectors there’s not a typo